common/supervisor/etc/supervisord.conf.d/postgresql.conf

```diff
@@ -1,6 +1,6 @@
 [program:postgresql]
 user=postgres
-command=/usr/lib/postgresql/12/bin/postgres -D "/opt/stellar/postgresql/data" -c config_file=/opt/stellar/postgresql/etc/postgresql.conf
+command=/usr/lib/postgresql/14/bin/postgres -D "/opt/stellar/postgresql/data" -c config_file=/opt/stellar/postgresql/etc/postgresql.conf
```

Comment:
@brahman81 @stellar/ops-team Do you foresee any issues with changing the version of Postgres that this image uses? I realize some folks use the quickstart image with persistent volumes that persist the data files for Postgres. It might be possible for us to keep with v12 by installing it from the Postgres apt repo, but v14 is the version we need to use if we use the Ubuntu packages.

Reply:
I'd need to double check to be sure but I think if you upgrade PG version it won't work out of the box with old datafiles. Keeping PG12 as workaround could work for now but at some stage you'll need to upgrade anyway so it's a problem we'll need to solve eventually.

Since we don't recommend quickstart for production usage maybe we could provide user-friendly way to let users know PG upgrade needs to happen and leave up to users to do? It should be possible to detect old data format during startup and print error message tell users they need to recreate the DB of perform upgrade. Do you think that's acceptable?

Reply:
I took a look into this. Postgres generates a rather nice message saying that the data files are the wrong version, and even points out what the version mismatch is. But unfortunately where it places that log won't show up in a good spot, and the only way for me to display that log in quickstart logs would be to display other noisy confusing postgres logs for users just using it normally.

Reply:
The quickstart image doesn't have a great "upgrade" story in other ways too. The supervisor config files are part of the persistent volume, which means that every time we've added a new service, or changed configs, folks wouldn't be getting the updates which would very likely create problems. With the upgrade to postgres 14 the config had to change, so it'd also fail to start postgres properly.

Reply:
I've pushed a change that prints out a nice error message now if the data folder is too old. The change is 48d06966. It looks like this:

```
postgres: 2023-04-19 04:32:19 UTC [18-1] FATAL:  database files are incompatible with server
postgres: 2023-04-19 04:32:19 UTC [18-2] DETAIL:  The data directory was initialized by PostgreSQL version 12, which is not compatible with this version 14.7 (Ubuntu 14.7-0ubuntu0.22.04.1).
```

The container then stops immediately, so this error is not easily missed.

Reply:
It looks good enough for dev usage, people can recreate the volume or try to upgrade manually.

Maybe, for clarity, you could add a note with potential solution for users? "Consider recreating the volume to start with empty database" or something along those lines?

Reply:
Unfortunately I don't control the message, it's output by postgres itself, and it could output other messages such as the data is corrupt or other things if there are other problems with the persistent volumes contents.